extensions/notebook/src/book/bookTreeItem.ts

```diff
@@ -134,13 +131,10 @@ export class BookTreeItem extends vscode.TreeItem {
 		let i = ++index;
 		while (i < this.book.tableOfContents.sections.length) {
 			let pathToNotebook: string;
-			if (this.book.version === BookVersion.v2 && (this.book.tableOfContents.sections[i] as IJupyterBookSectionV2).file) {
+			if (this.book.tableOfContents.sections[i].file) {
 				// The Notebook editor expects a posix path for the resource (it will still resolve to the correct fsPath based on OS)
-				pathToNotebook = path.posix.join(this.book.root, (this.book.tableOfContents.sections[i] as IJupyterBookSectionV2).file.concat('.ipynb'));
-			} else if ((this.book.tableOfContents.sections[i] as IJupyterBookSectionV1).url) {
-				pathToNotebook = path.posix.join(this.book.root, content, (this.book.tableOfContents.sections[i] as IJupyterBookSectionV1).url.concat('.ipynb'));
+				pathToNotebook = this.book.version === BookVersion.v2 ? path.posix.join(this.book.root, this.book.tableOfContents.sections[i].file.concat('.ipynb')) : path.posix.join(this.book.root, content, this.book.tableOfContents.sections[i].file.concat('.ipynb'));
```

Comment:
You could separate out the ipynb concat call into another variable to shorten this ternary operator.

Reply:
Same comment for the other concat operation above this one. Long ternary operators can be confusing to read at a glance.